# Kaggle competition

New Data available, in different format so we need to repreprocess everything

# 0 - Imports

In [2]:
import pandas as pd
import numpy as np
import torch

# I - Data Preprocessing

In [5]:
df_train = pd.read_csv('train_submission.csv')
df_train.head(2)

ID   Usage                                               Text Label
0  136  Public  Finalment  Atena le recibe en l'acropoli d'Ate...   arg
1   62  Public  Jane Laffort fille de Joseph Laffort et d' Ang...   lat

In [6]:
df_test = pd.read_csv('test_without_labels.csv')
df_test.head(2)

ID    Usage                                               Text
0  55  Private           Ponovo dobija riječni oblik do Drežnice.
1  71  Private  Se formaron aproximadamente hace apenas unos 1...

In [11]:
print("Train Shape = ",df_train.shape)
print("Test shape = ",df_test.shape)
print("List labels length = ", len(df_train['Label'].unique()))

Train Shape =  (38854, 4)
Test shape =  (38827, 3)
List labels length =  390


Ok - So we have 38K sentences in different languages, to classify in 390 categories. If the class is balanced, this would represent a 100:1 ratio, so ok to train without generating new sentences I assume. So first baseline algo will just train an NLP classifier on train dataset, and then use the test dataset to see how good it actually is.

# II - Bert